overlord/restart/restart.go

```diff
@@ -362,3 +362,139 @@ func ReplaceBootID(st *state.State, bootID string) {
 	rm := restartManager(st, "internal error: cannot mock a restart request before RestartManager initialization")
 	rm.bootID = bootID
 }
+
+// markTaskForRestart sets certain properties on a task to mark it for system restart.
+// The status argument is the status that the task will have after the system restart.
+func markTaskForRestart(t *state.Task, status state.Status) {
+	// XXX: Preserve previous restart behaviour for classic in the undo cases, is this still
+	// necessary?
+	if release.OnClassic && (status == state.UndoStatus || status == state.UndoneStatus) {
+		t.SetStatus(status)
+		t.Logf("Skipped automatic system restart on classic system when undoing changes back to previous state")
+		return
+	}
+
+	rm := restartManager(t.State(), "internal error: cannot request a restart before RestartManager initialization")
+
+	// store current boot id to be able to check later if we have rebooted or not
+	t.Set("wait-for-system-restart-from-boot-id", rm.bootID)
+	t.SetToWait(status)
+	setWaitForSystemRestart(t.Change())
+
+	t.Logf("Task set to wait until a system restart allows to continue")
+}
+
+// changeRestartParameters returns either existing restart parameters from a
+// previous reboot request, or a new empty instance of RestartParameters which
+// needs to be init().
+func changeRestartParameters(chg *state.Change) (*RestartParameters, error) {
```

Comment:
in this name is a bit unclear if change is a name or a verb. Maybe restartParametersFromChange is a better and follow some other names we already have (DeviceCtxFromState).

Reply:
good point!